docs/application-tutorial/game_of_life.py

```diff
@@ -0,0 +1,551 @@
+        convoluted_grid = by_hand_padding(grid, convoluted_grid)
+
+    grid_a = table_cp_next_cell_4b_a[convoluted_grid]
+    grid = grid_a + grid
+    grid = table_cp_next_cell_4b_b[grid]
+
+    return grid
+
+
+@fhe.compiler({"grid": "encrypted"})
+def update_grid_method_5b(grid):
+
+    # Method which uses a single TLU of 5 bits
+
+    # This is to workaround the fact that we have no pad option in fhe.conv
+    do_padded_fix = True
+
+    convoluted_grid = fhe.conv(
+        grid,
+        weights_method_5b.reshape(1, 1, *weights_method_5b.shape),
+        strides=(1, 1),
+        dilations=(1, 1),
+        group=1,
+        pads=(1, 1, 1, 1) if not do_padded_fix else (0, 0, 0, 0),
+    )
+
+    if do_padded_fix:
+        convoluted_grid = by_hand_padding(grid, convoluted_grid)
+
+    grid = table_cp_next_cell_5b[convoluted_grid]
+
+    return grid
+
+
+@fhe.compiler({"grid": "encrypted"})
+def update_grid_basic(grid):
+
+    # Method which follows the naive approach
+
+    # This is to workaround the fact that we have no pad option in fhe.conv
+    do_padded_fix = True
+
+    convoluted_grid = fhe.conv(
+        grid,
+        weights_method_basic.reshape(1, 1, *weights_method_basic.shape),
+        strides=(1, 1),
+        dilations=(1, 1),
+        group=1,
+        pads=(1, 1, 1, 1) if not do_padded_fix else (0, 0, 0, 0),
+    )
+
+    if do_padded_fix:
+        convoluted_grid = by_hand_padding(grid, convoluted_grid)
+
+    grid = table_cp_next_cell_basic_a[convoluted_grid] | (
+        table_cp_next_cell_basic_b[convoluted_grid] & (grid == 1)
+    )
+
+    return grid
+
+
+# Function for Game of Life
+def update_grid(grid, method="method_3b"):
+
+    assert grid.ndim == 4
+
+    if method == "method_basic":
+        res = update_grid_basic(grid)
+
+    if method == "method_3b":
+        res = update_grid_method_3b(grid)
+
+    if method == "method_4b":
+        res = update_grid_method_4b(grid)
+
+    if method == "method_5b":
+        res = update_grid_method_5b(grid)
+
+    return res
+
+
+# Graphic functions
+# The graphical functions of this code were inspired by those of
+# https://github.com/matheusgomes28/pygame-life/blob/main/pygame_life.py
+def manage_graphics_and_refresh(
+    grid,
+    count,
+    dimension,
+    nb_initial_points,
+    border_size,
+    screen,
+    background_refresh_color,
+    background_color,
+    life_color,
+    time_new_grid_sleep,
+    time_sleep,
+    refresh_every,
+    do_text_output,
+):
+    make_new_grid = False
+
+    if count == 0:
+        make_new_grid = True
+    elif refresh_every > 0:
+        if (count % refresh_every) == 0:
+            make_new_grid = True
+
+    count += 1
+
+    # Refresh the grid from time to time
+    if make_new_grid:
+        grid = np.random.randint(2, size=(1, 1, dimension, dimension), dtype=np.int8)
+        screen.fill(background_refresh_color)
+        pygame.display.flip()
+        time.sleep(time_new_grid_sleep)
+
+        screen.fill(background_color)
```

Comment:
no need for else
always do screen.fill(background_color)

Reply:
ok done, I always do